xbmc/storage/DetectDVDType.h

```diff
@@ -62,16 +64,17 @@ class CDetectDVDMedia : public CThread
 private:
   static CCriticalSection m_muReadingMedia;
 
-  static int m_DriveState;
+  static DriveState m_DriveState;
```

Comment:
Missing initialization?

Reply:
Don't think so in this case:
```
[build] FAILED: build/profiles/CMakeFiles/profiles.dir/ProfileManager.cpp.o 
[build] /usr/bin/ccache /usr/bin/c++  -I/home/arch/Github/xbmc -I/home/arch/Github/xbmc/lib -I/home/arch/Github/xbmc/xbmc -I/home/arch/Github/xbmc/xbmc/platform/linux -I/home/arch/Github/xbmc/xbmc/cores/VideoPlayer -I/home/arch/Github/xbmc/build/build -I/home/arch/Github/xbmc/xbmc/platform/posix -isystem /home/arch/Github/xbmc/build/build/include -isystem /usr/include/dbus-1.0 -isystem /usr/lib/dbus-1.0/include -isystem /usr/include/python3.10 -isystem /usr/include/samba-4.0 -isystem /usr/include/libxml2 -isystem /home/arch/Github/xbmc/build/build/cores/RetroPlayer/messages -isystem /usr/include/freetype2 -isystem /usr/include/fribidi -isystem /home/arch/Github/xbmc/xbmc/contrib -isystem /home/arch/Github/xbmc/build/build/libdvd/include -isystem /usr/include/lzo -isystem /usr/include/libdrm -Wall -Wdouble-promotion -Wmissing-field-initializers -Wsign-compare -Wextra -Wno-unused-parameter -Wno-cast-function-type -Wnon-virtual-dtor -g -D_DEBUG -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -DHAVE_SSE=1 -DHAVE_SSE2=1 -DHAVE_SSE3=1 -DHAVE_SSSE3=1 -DHAVE_SSE4_1=1 -D__STDC_CONSTANT_MACROS -D_FILE_OFFSET_BITS=64 -DHAS_POSIX_NETWORK -DHAS_LINUX_NETWORK -DHAS_BUILTIN_SYNC_ADD_AND_FETCH=1 -DHAS_BUILTIN_SYNC_SUB_AND_FETCH=1 -DHAS_BUILTIN_SYNC_VAL_COMPARE_AND_SWAP=1 -DHAVE_INOTIFY=1 -DHAVE_POSIX_FADVISE=1 -DHAVE_LOCALTIME_R=1 -DHAVE_GMTIME_R=1 -DHAVE_INTTYPES_H=1 -DHAS_ALSA=1 -DHAS_AVAHI=1 -DHAS_ZEROCONF=1 -DHAVE_LIBBLURAY=1 -DHAVE_LIBBLURAY_BDJ=1 -DHAVE_LIBCEC=1 -DHAS_DBUS=1 -DHAS_ISO9660PP=1 -DHAVE_LCMS2=1 -DCMS_NO_REGISTER_KEYWORD=1 -DHAS_LIRC=1 -DHAS_WEB_SERVER=1 -DHAS_WEB_INTERFACE=1 -DHAS_FILESYSTEM_NFS=1 -DHAS_NFS_SET_TIMEOUT -DHAS_NFS_MOUNT_GETEXPORTS_TIMEOUT -DHAS_AIRPLAY=1 -DHAS_PULSEAUDIO=1 -DHAS_PYTHON=1 -DHAS_FILESYSTEM_SMB=1 -DHAVE_LIBUDEV=1 -DHAS_UDFREAD=1 -DHAVE_LIBXSLT=1 -DHAVE_LIBVA=1 -DHAS_GLX=1 -DHAVE_LIBVDPAU=1 -DFFMPEG_VER_SHA=\"4.4.1\" -DSPDLOG_FMT_EXTERNAL -DSPDLOG_DEBUG_ON -DSPDLOG_NO_ATOMIC_LEVELS -DSPDLOG_ENABLE_PATTERN_PADDING -DHAS_EGL=1 -DHAVE_X11=1 -DHAVE_LIBXRANDR=1 -DHAVE_HDR_OUTPUT_METADATA=1 -DHAVE_DRM_MODIFIER_NAME=1 -DHAS_GL=1 -DHAS_MYSQL=1 -DHAS_UPNP=1 -DHAS_DVD_DRIVE -DHAS_CDDA_RIPPER -DHAS_AIRTUNES=1 -DBIN_INSTALL_PATH=\"/usr/local/lib/kodi\" -DINSTALL_PATH=\"/usr/local/share/kodi\" -Werror=double-promotion -Werror=missing-field-initializers -Werror=sign-compare -std=c++17 -MD -MT build/profiles/CMakeFiles/profiles.dir/ProfileManager.cpp.o -MF build/profiles/CMakeFiles/profiles.dir/ProfileManager.cpp.o.d -o build/profiles/CMakeFiles/profiles.dir/ProfileManager.cpp.o -c /home/arch/Github/xbmc/xbmc/profiles/ProfileManager.cpp
[build] In file included from /home/arch/Github/xbmc/xbmc/profiles/ProfileManager.cpp:42:
[build] /home/arch/Github/xbmc/xbmc/storage/DetectDVDType.h:66:21: error: ISO C++ forbids in-class initialization of non-const static member ‘MEDIA_DETECT::CDetectDVDMedia::m_DriveState’
[build]    66 |   static DriveState m_DriveState{DriveState::DRIVE_NONE};
```

Reply:
it is already initialized here https://github.com/xbmc/xbmc/blob/fffd1d76ee5988681445de596bdea8ed2397bee7/xbmc/storage/DetectDVDType.cpp#L34
enen, you changed that intuitively